src/components/Publish/Navigation/index.tsx

```diff
@@ -1,21 +1,53 @@
 import { FormikContextType, useFormikContext } from 'formik'
-import React, { ReactElement } from 'react'
+import React, { ReactElement, useEffect } from 'react'
+import { useRouter } from 'next/router'
 import { FormPublishData } from '../_types'
 import { wizardSteps } from '../_constants'
 import styles from './index.module.css'
 
 export default function Navigation(): ReactElement {
+  const router = useRouter()
+
   const {
     values,
     errors,
     touched,
     setFieldValue
   }: FormikContextType<FormPublishData> = useFormikContext()
 
+  const queryString = window.location.search
```

Comment:
seems to all work but just wondering if there's a way to do same thing without resorting to `window` (as this might conflict with some server-side rendering) and without having to sync more values between different places. 

Just thinking out loud, we already have a routing system going on with Next.js so we should be able to end up with URLs like `/publish/1` by using nested routes within `src/pages/`: https://nextjs.org/docs/routing/introduction

With that in place, the whole screen switching mechanism could be switched to then getting the screen page from `router.pathname` instead of using `values.user.stepCurrent` and we would not have to sync this form state with the URL query params.